junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -840,6 +842,25 @@ private static void findNestedClasses(Class<?> clazz, Set<Class<?>> candidates)
 	}
 
 	/**
+	 * Method comparator based upon JUnit4 org.junit.internal.MethodSorter implementation.
+	 */
+	private static int defaultMethodSorter(Method method1, Method method2) {
+		if (method1 == method2) {
+			return 0;
+		}
+		String name1 = method1.getName();
+		String name2 = method2.getName();
+		int comparison = Integer.compare(name1.hashCode(), name2.hashCode());
+		if (comparison == 0) {
+			comparison = name1.compareTo(name2);
+			if (comparison == 0) {
+				comparison = method1.toString().compareTo(method2.toString());
+			}
+		}
+		return comparison;
+	}
```

Comment:
Oh whoops, I hadn't tested it in my IDE!

How about this; does this compile for you?

```java
return Comparator.comparingInt((Method method) -> method.getName().hashCode())
        .thenComparing(Method::getName)
        .thenComparing(Method::toString)
        .compare(method1, method2);
```